MekHQ/src/mekhq/campaign/personnel/Person.java

```diff
@@ -2228,13 +2209,27 @@ public static Person generateInstanceFromXML(Node wn, Campaign c, Version versio
                         retVal.formerSpouses.add(FormerSpouse.generateInstanceFromXML(wn3));
                     }
                 } else if (wn2.getNodeName().equalsIgnoreCase("dueDate")) {
-                    SimpleDateFormat df = new SimpleDateFormat("yyyy-MM-dd hh:mm:ss");
-                    retVal.dueDate = (GregorianCalendar) GregorianCalendar.getInstance();
-                    retVal.dueDate.setTime(df.parse(wn2.getTextContent().trim()));
+                    if (version.isLowerThan("0.47.6")) {
+                        try {
+                            retVal.dueDate = LocalDate.parse(wn2.getTextContent().trim(),
+                                    DateTimeFormatter.ofPattern("yyyy-MM-dd hh:mm:ss"));
+                        } catch (Exception ignored) {
+                        }
+                    } else {
+                        retVal.dueDate = LocalDate.parse(wn2.getTextContent().trim(),
+                                DateTimeFormatter.ofPattern(DATE_SAVE_FORMAT));
+                    }
                 } else if (wn2.getNodeName().equalsIgnoreCase("expectedDueDate")) {
-                    SimpleDateFormat df = new SimpleDateFormat("yyyy-MM-dd hh:mm:ss");
-                    retVal.expectedDueDate = (GregorianCalendar) GregorianCalendar.getInstance();
-                    retVal.expectedDueDate.setTime(df.parse(wn2.getTextContent().trim()));
+                    if (version.isLowerThan("0.47.6")) {
```

Comment:
Is there a way to just parse a date without specifying the exact format? Seems like a built-in library should be smart enough to contain that logic already.

Reply:
I didn't find one while researching, annoyingly.

Reply:
You can but it fails on machines with different cultures from what wrote the date, can't pass around CPNX's.

Reply:
All right, in that case, let's consolidate this logic into a separate function.

Reply:
I plan on doing so for all of the personnel migration cases, but that will be in a different PR.